ccdproc/combiner.py

```diff
@@ -2,52 +2,89 @@
 
 """This module implements the combiner class."""
 
-import numpy as np
-from numpy import ma
-
 try:
     import bottleneck as bn
 except ImportError:
     HAS_BOTTLENECK = False  # pragma: no cover
 else:
     HAS_BOTTLENECK = True
 
+import array_api_compat
+import array_api_extra as xpx
 from astropy import log
 from astropy.nddata import CCDData, StdDevUncertainty
 from astropy.stats import sigma_clip
 from astropy.utils import deprecated_renamed_argument
+from numpy import mgrid as np_mgrid
+from numpy import ravel_multi_index as np_ravel_multi_index
 
 from .core import sigma_func
 
 __all__ = ["Combiner", "combine"]
 
 
-def _default_median():  # pragma: no cover
+def _default_median(xp=None):  # pragma: no cover
     if HAS_BOTTLENECK:
```

Comment:
Does Bottleneck work with the array api? If not, would it be better to change the order of this logic and only use bottleneck if the namespace is numpy / None?

Reply:
Yeah, good call. I can check later what libraries bottlneck works with. Mostly I don't want to change the behavior for current users

Reply:
Opened an issue to track this: https://github.com/astropy/ccdproc/issues/904